.github/workflows/main.yml

```diff
@@ -217,6 +217,25 @@ jobs:
           path-to-profile: coverage.txt
           parallel: true
 
+  ###############################
+  # run unit tests for submodules
+  ###############################
+  submodule-unit-test:
```

Comment:
Can we just add `- unit-module` to the matrix in the job above?

Reply:
Done!